### PR TITLE
dep(nbconvert): major version upgrade

### PR DESCRIPTION
**Pull request by josephmcasey**
# `dep/nbconvert`

Resolves https://github.com/danielfrg/mkdocs-jupyter/issues/121
Resolves https://github.com/danielfrg/mkdocs-jupyter/issues/126
Resolves https://github.com/danielfrg/mkdocs-jupyter/issues/109

## Changelog Details

- fix(nbconvert2): monkey patch partially functional clean_html
- test(renders): replaces subprocess to better enable hot-reloading breakpoint debugging
- dep(nbconvert): upgrade nbconvert 6.5.0 to 7.2.9
- style(fmt): changes produced by fmt cmd in Taskfile.yml
- build(poetry.lock): refresh file hashes with poetry v1.4.0
- build(gitignore): ignore files generated by mkdocs serve

## Resources
- https://github.com/jupyter/nbconvert/issues/1894

### CODE REVIEW

File: mkdocs_jupyter/nbconvert2.py [hunk 1]
```diff
@@ -240,9 +248,9 @@ def custom_markdown2html(source):
     This is done so it's the same HTML structure that for regular non-language
     sections.
     """
-    return MarkdownWithMath(renderer=CustomMarkdownRendered(escape=False)).render(
-        source
-    )
+    return MarkdownWithMath(
+        renderer=CustomMarkdownRendered(escape=False)
+    ).render(source)
```
Comment (josephmcasey) [inline mkdocs_jupyter/nbconvert2.py:253]: style(fmt): changes produced by `fmt` cmd in `Taskfile.yml`

File: mkdocs_jupyter/tests/mkdocs/docs/demo-script.py [hunk 2]
```diff
@@ -117,7 +117,7 @@
 
 # %%
 x = linspace(0, 5, 10)
-y = x ** 2
+y = x**2
```
Comment (josephmcasey) [inline mkdocs_jupyter/tests/mkdocs/docs/demo-script.py:120]: style(fmt): changes produced by `fmt` cmd in `Taskfile.yml`

File: mkdocs_jupyter/tests/test_base_usage.py [hunk 3]
```diff
@@ -1,34 +1,36 @@
...
+
+def test_notebook_renders(input):
+    filename, renders = input
+
     this_dir = os.path.dirname(os.path.realpath(__file__))
-    mkdocs_dir = os.path.join(this_dir, "mkdocs")
+    config_file = os.path.join(this_dir, f"mkdocs/{filename}")
+
+    try:
+        build(load_config(config_file))
```
Comment (josephmcasey) [inline mkdocs_jupyter/tests/test_base_usage.py:30]: test(renders): replaces subprocess to better enable hot-reloading breakpoint debugging

File: pyproject.toml [hunk 4]
```diff
@@ -22,7 +22,7 @@ include = ["mkdocs_jupyter/**/assets/*"]
 
 [tool.poetry.dependencies]
 python = ">=3.7.1,<4"
-nbconvert = "^6.2.0"
+nbconvert = "^7.2.9"
```
Comment (josephmcasey) [inline pyproject.toml:25]: dep(nbconvert): upgrade nbconvert 6.5.0 to 7.2.9

File: mkdocs_jupyter/nbconvert2.py [hunk 5]
```diff
@@ -99,6 +100,13 @@ def nb2html(
         # "markdown2html": custom_markdown2html,
     }
 
+    # Ensures SVG rendered with data and rdkit exit with NOOP
+    # Source: https://github.com/jupyter/nbconvert/issues/1894#issuecomment-1334355109
+    def custom_clean_html(element):
+        return element.decode() if isinstance(element, bytes) else str(element)
+
+    default_filters["clean_html"] = custom_clean_html
```
Comment (josephmcasey) [inline mkdocs_jupyter/nbconvert2.py:108]: fix(nbconvert2): monkey patch partially functional clean_html

File: poetry.lock [hunk 6]
```diff
@@ -696,56 +1244,85 @@ pymdown-extensions = ">=9.0"
...
-sphinx = ["mock", "moto", "myst-parser", "Sphinx (>=1.7)", "sphinx-book-theme"]
-test = ["black", "check-manifest", "flake8", "ipykernel", "ipython (<8.0.0)", "ipywidgets (<8.0.0)", "mypy", "pip (>=18.1)", "pre-commit", "pytest (>=4.1)", "pytest-asyncio", "pytest-cov (>=2.6.1)", "setuptools (>=60.0)", "testpath", "twine (>=1.11.0)", "xmltodict"]
+dev = ["pre-commit"]
+docs = ["autodoc-traits", "mock", "moto", "myst-parser", "nbclient[test]", "sphinx (>=1.7)", "sphinx-book-theme"]
+test = ["ipykernel", "ipython", "ipywidgets", "nbconvert (>=7.0.0)", "pytest (>=7.0)", "pytest-asyncio", "pytest-cov (>=4.0)", "testpath", "xmltodict"]
 
 [[package]]
 name = "nbconvert"
-version = "6.5.0"
+version = "7.2.9"
```
Comment (josephmcasey) [inline poetry.lock:1306]: build(poetry.lock): refresh file hashes with poetry v1.4.0

**Comment by danielfrg** [comment]
Thank you so much for the PR!
I will merge because it's a poetry issue and I am planning to move to hatch.

**Pull request merged**